### PR TITLE
revert: remove enable_stream_row_count config #10261

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
Interesting. May I ask for the motivation?

**Comment by lmatz** [comment]
With 500+ MV on a single machine, one user's Prometheus takes 16GB.

**Comment by lmatz** [comment]
I am thinking just giving MV and sinks two dedicated metrics, or judging `actor_id_string` or other similar things to enable row_executor_count just for MV and sinks.

**Comment by lmatz** [comment]
SCR-20230817-u6l

Added a `sink_info` metric into Prom so that we can get the actor_id -> sink_name when building the panel

File: src/stream/src/executor/wrapper/trace.rs [hunk 1]
```diff
@@ -38,14 +39,16 @@ pub async fn trace(
 
     let span_name = pretty_identity(&info.identity, actor_id, executor_id);
 
+    let is_sink_or_mv = info.identity.contains("Materialize") || info.identity.contains("Sink");
```
Comment (BugenZhao) [inline src/stream/src/executor/wrapper/trace.rs:42]: This seems fragile but I'm unsure if there's a better way.

**Review by BugenZhao — APPROVED** [review]
LGTM. PTAL @hzxa21 🥰

**Review by zwang28 — APPROVED** [review]

File: src/stream/src/executor/wrapper/trace.rs (hunk 1 above)
Comment (lmatz) [reply src/stream/src/executor/wrapper/trace.rs:42]: let me open an issue to track this, a quick fix for Kaito

Comment (hzxa21) [reply src/stream/src/executor/wrapper/trace.rs:42]: +1. It seems a little bit hacky here.

**Review by hzxa21 — APPROVED** [review]

**Comment by zwang28** [comment]
After this PR, the resulted file size reduces from 600MB to 100MB:
`curl -v http://127.0.0.1:[cn prometheus port]/api/v1/query`

Though it's still large, at least grafana can present it now.

**Comment by zwang28** [comment]
Storage metric takes up 80%.
```
zzzzz:~ z28wang$ grep "store" cn.prom_query.2.log |wc -l
 4236203
zzzzz:~ z28wang$ grep "stream" cn.prom_query.2.log |wc -l
 1435195
zzzzz:~ z28wang$ cat cn.prom_query.2.log | wc -l
 5821215
zzzzz:~ z28wang$ du -h cn.prom_query.2.log
464M	cn.prom_query.2.log
```

**Pull request merged**